app/src/CMakeLists.txt

```diff
@@ -0,0 +1,13 @@
+target_include_directories(${PROJECT_NAME}
+    PRIVATE
+        ${CMAKE_CURRENT_SOURCE_DIR}
+)
+
+target_sources(${PROJECT_NAME}
+    PRIVATE
+        ${CMAKE_CURRENT_SOURCE_DIR}/main.cpp
+
+        ${CMAKE_CURRENT_SOURCE_DIR}/CMakeLists.txt
```

Comment:
I've never seen something like that before. Why is that a target source?

Reply:
This is just a convention required to include and index these files in IDEs optionally used as generators in cmake call (Visual Studio, QtCreator ...). Personally, I'm fine that anything that is code which affects the final binaries is included.